systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -139,6 +145,85 @@ void testKafkaNodePoolBrokerIdsManagementUsingAnnotations() {
             KafkaNodePoolUtils.getCurrentKafkaNodePoolIds(testStorage.getNamespaceName(), nodePoolNameB).equals(Arrays.asList(0, 2, 3, 5)));
     }
 
+    /**
+     * @description This test case verifies changing of roles in Kafka Node Pools.
+     *
+     * @steps
+     *  1. - Deploy a Kafka instance with annotations to manage Node Pools and Initial 2 NodePools, both with mixed role, first one stable, second one which will be modified.
+     *  2. - Create KafkaTopic with replica number requiring all Kafka Brokers to be present.
+     *  3. - Annotate one of Node Pools to perform manual Rolling Update.
+     *     - Rolling Update started.
+     *  3. - Change role of Kafka Node Pool from mixed to controller only role.
+     *     - Role Change is being prevented because a previously created KafkaTopic still has some replicas present on the node to be scaled down, also there is original Rolling Update going on.
+     *  4. - Original Rolling Update finishes successfully.
+     *  5. - Delete previously created KafkaTopic.
+     *     - KafkaTopic is deleted, and roll of Node Pool whose role was changed begins resulting in new nodes with expected role.
+     *  6. - Change role of Kafka Node Pool from controller only to mixed role.
+     *     - Kafka Node Pool changes role to mixed role.
+     *  7. - Produce and consume messages on newly created KafkaTopic with replica count requiring also new brokers to be present.
+     *
+     * @usecase
+     *  - kafka-node-pool
+     */
+    @ParallelNamespaceTest
+    void testNodePoolsRolesChanging() {
+        assumeTrue(Environment.isKRaftModeEnabled());
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+        final Kafka kafka = KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 1, 1).build();
+
+        // Stable Node Pool that serves only for having at least 1 broker still available
+        final String stablePoolName = testStorage.getKafkaNodePoolName() + "-stable";
+        final KafkaNodePool stablePool = KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(stablePoolName, kafka, 3).build();
+
+        // node pool with role which will be changed, starting with having mixed role
+        final String volatileRolePoolName = testStorage.getKafkaNodePoolName() + "-volatile";
+        final KafkaNodePool volatilePool = KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(volatileRolePoolName, kafka, 3).build();
+        final LabelSelector volatilePoolLabelSelector = KafkaNodePoolResource.getLabelSelector(testStorage.getClusterName(), volatileRolePoolName, ProcessRoles.CONTROLLER);
+
+        LOGGER.info("Deploy 2 KafkaNodePools {}, {}, in {}", stablePoolName, volatileRolePoolName, testStorage.getNamespaceName());
```

Comment:
You are also deploying Kafka. Plus this log is in `ResourceManager`, so you don't need to put it here (I mean info that the resources are being deployed)